raft/raftpb/raft.proto

```diff
@@ -64,6 +64,7 @@ message Message {
 	optional Snapshot    snapshot    = 9  [(gogoproto.nullable) = false];
 	optional bool        reject      = 10 [(gogoproto.nullable) = false];
 	optional uint64      rejectHint  = 11 [(gogoproto.nullable) = false];
+	optional bytes       context     = 12;
```

Comment:
I prefer defining a campaign enum and use a campaign type here instead of context.